### PR TITLE
locale messages helpers (insert, edit existing, create first)

### PR DESCRIPTION
**Pull request by chvister**

### CODE REVIEW

File: packages/react-lowcode/src/localization/json.ts [hunk 1]
```diff
@@ -27,16 +32,33 @@ export function parseLocaleAST(ast: SourceFile, languageLocale = "en") {
...
 // it preserves original JSON formatting
-export function patchLocaleJSON(localeFile: string, changedMessages: Message[], originalMessages: Message[]) {
-  for (let i = changedMessages.length; i >= 0; i--) {
-    if (changedMessages[i]?.value == originalMessages[i]?.value) {
-      console.log("Equal")
-    } else {
-      const before = localeFile.substring(0, originalMessages[i].position.pos + 1)
-      const after = localeFile.substring(originalMessages[i].position.end - 1)
-      localeFile = before + changedMessages[i].value + after
+export function patchLocaleJSON(
```
Comment (jozef-slezak) [inline packages/react-lowcode/src/localization/json.ts:35]: I am thinking:
- you can rename this function to `putLocaleMessage()` and keep the original function `patchLocaleJSON` for backward compatibility
- we have duplicit parameters in the function `originalMessages` and `localeFile` - what if we calculate `originalMessages` if not set (we could allow not to pass this duplicit argument)

**Comment by chvister** [comment]
@jozef-slezak so:
- created new function `putLocaleMessage()`
- origin function kept `patchLocaleJSON()`
- `JSON.stringify()` for escaped characters working properly 
- dedleted duplicity `originalMessages`

File: packages/react-lowcode/src/localization/messageFunctions.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        0,
+        originalMessages[i].position.end
+      )
+      const after = originalLocaleStringJSON?.substring(
+        originalMessages[i].position.end
+      )
+
+      const space = copyRow?.match(/([\s]+)/g)
+      const newRow = `${space}${JSON.stringify(messageId)}: ${JSON.stringify(newValue)}`
+      originalLocaleStringJSON = `${before},` + newRow + after
```
Comment (jozef-slezak) [inline packages/react-lowcode/src/localization/messageFunctions.ts:65]: you should not rewrite value in original variable

File: packages/react-lowcode/src/localization/messageFunctions.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  for (let i = originalMessages.length - 1; i >= 0; i--) {
+    if (originalMessages[i].id === messageId) {
+      const before = originalLocaleStringJSON?.substring(
+        0,
+        originalMessages[i].position.pos + 1
+      )
+      const after = originalLocaleStringJSON?.substring(
+        originalMessages[i].position.end 
+      )
+      originalLocaleStringJSON = before + JSON.stringify(newValue) + after
```
Comment (jozef-slezak) [inline packages/react-lowcode/src/localization/messageFunctions.ts:25]: you should not rewrite value in original variable

File: packages/react-lowcode/src/localization/json.ts [hunk 4]
```diff
@@ -27,16 +41,51 @@ export function parseLocaleAST(ast: SourceFile, languageLocale = "en") {
...
+export function putLocaleMessage(
+  messageId: string,
+  newValue: string,
+  originalLocaleStringJSON: string | undefined
+) {
+  const originalMessages =
+    parseLocaleJSON(originalLocaleStringJSON as string) || []
+  const found = originalMessages?.find((message) => message.id == messageId)
+
+  if (originalMessages?.length == 0) {
```
Comment (jozef-slezak) [inline packages/react-lowcode/src/localization/json.ts:74]: @chvister, nice work. Just last few commetns:

please move/extract following logic to a function `newFunction(changedMessage: Message,  originalMessages: Message[])` and use it in both existing functions:
* putMessage
* patchLocaleJSON (for backward compatibility)

**Pull request merged**